hive/src/main/java/io/confluent/connect/storage/hive/HiveSchemaConverter.java

```diff
@@ -90,6 +98,29 @@ public static TypeInfo convertMap(Schema schema) {
   }
 
   public static TypeInfo convertPrimitive(Schema schema) {
-    return TYPE_TO_TYPEINFO.get(schema.type());
+    if (schema.name() == null) {
+      return TYPE_TO_TYPEINFO.get(schema.type());
+    }
+
+    //TODO: should take Hive version into consideration
+    switch (schema.name()) {
+      case Decimal.LOGICAL_NAME:
+        String scale = schema.parameters().get(Decimal.SCALE_FIELD);
+        String precision = schema.parameters().getOrDefault(CONNECT_AVRO_DECIMAL_PRECISION_PROP,
+            CONNECT_AVRO_DECIMAL_PRECISION_DEFAULT);
+        return new DecimalTypeInfo(Integer.parseInt(precision), Integer.parseInt(scale));
```

Comment:
Is there any value for precision and/or scale that requires different type than `DecimalTypeInfo` ?

Reply:
You mean "is there any type other than `DecimalTypeInfo` that requires precision and scale"? No. This is the only logical type that requires it.